### ISSUE TITLE
[BUG][code-analyzer] regex:custom_rules won't run

### ISSUE THREAD

**Issue by Damecek** [forcedotcom/code-analyzer#1818]
### Have you tried to resolve this issue yourself first?

- [x] I confirm I have gone through the above steps and still have an issue to report

### Bug Description

The custom rules for regex engine in the `code-analyzer.yml` won't run. Not even the example one from the generated `code-analyzer.yml`.

### Output / Logs

```
/usr/local/lib/sf/bin/sf code-analyzer run -c code-analyzer.yml -t force-app/main/default/classes/test.cls

Streaming logs in real time to:
    /var/folders/zv/53f3cdqn4wd0bsry_61xj0_c0000gn/T/sfca-2025_05_22_13_05_05_504.log

Selecting rules... done.
Executing rules... done. Executed rules from regex, pmd, cpd.

Violation file paths relative to '/Users/adam/IdeaProjects/salesforce/force-app/main/default/classes'.
 # Severity     Rule                       Location      Message                                                  
 ─ ──────────── ────────────────────────── ───────────── ──────────────────────────────────────────────────────── 
 1 3 (Moderate) pmd:ClassNamingConventions test.cls:1:21 The class name 'test' doesn't match '[A-Z][a-zA-Z0-9_]*' 
 2 5 (Info)     pmd:ApexDoc                test.cls:1:21 Missing ApexDoc comment                                  


=== Summary

Found 2 violation(s) across 1 file(s):
    1 Moderate severity violation(s) found.
    1 Info severity violation(s) found.

Additional log information written to:
    /var/folders/zv/53f3cdqn4wd0bsry_61xj0_c0000gn/T/sfca-2025_05_22_13_05_05_504.log
```

### Steps To Reproduce

have this in your `code-analyzer.yml`
```yml
    regex:
        disable_engine: false

        # ---- [Example usage]: ---------------------
        # engines:
        #   regex:
        #     custom_rules:
        #       "NoTodoComments":
        #         regex: /\/\/[ \t]*TODO/gi
        #         file_extensions: [".apex", ".cls", ".trigger"]
        #         description: "Prevents TODO comments from being in apex code."
        #         violation_message: "A comment with a TODO statement was found. Please remove TODO statements from your apex code."
        #         severity: "Info"
        #         tags: ["TechDebt"]
        # -------------------------------------------
        custom_rules:
            "NoTodoComments":
                regex: /\/\/[ \t]*TODO/gi
                file_extensions: [ ".apex", ".cls", ".trigger" ]
                description: "Prevents TODO comments from being in apex code."
                violation_message: "A comment with a TODO statement was found. Please remove TODO statements from your apex code."
                severity: "Info"
                tags: [ "TechDebt" ]
```
and run on class with todo comment like this.
```apex
public with sharing class test {
    // todo: asd
}
```

`/usr/local/lib/sf/bin/sf code-analyzer run -c code-analyzer.yml -t force-app/main/default/classes/test.cls`

### Expected Behavior

the `NoTodoComments` should appear in the result

### Operating System

macos 15.5 (24F74)

### Salesforce CLI Version

@salesforce/cli/2.89.8 darwin-arm64 node-v22.15.0

### Code Analyzer Plugin (code-analyzer) Version

code-analyzer 5.0.0

### Node Version

v22.15.0

### Java Version

 openjdk version "23.0.2" 2025-01-21 OpenJDK Runtime Environment Homebrew (build 23.0.2) OpenJDK 64-Bit Server VM Homebrew (build 23.0.2, mixed mode, sharing)

### Python Version

Python 3.10.11

### Urgency

High

**Comment by stephen-carter-at-sf** [comment]
As stated in the the custom_rules doc at https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/engine-regex.html

```
'tags' - [Optional] The string array of tag values to apply to this rule by default.
             Default: ['Recommended']
```
That is when you define a regex rule, if you do not add in the "tags" field then by default it will have `['Recommended']`. So if you want rules to run by default, you don't need to manually specify tags at all.

https://github.com/forcedotcom/code-analyzer-core/blob/dev/packages/code-analyzer-regex-engine/src/config.ts#L47
